lib/src/mustachio/renderer_base.dart

```diff
@@ -183,10 +191,21 @@ abstract class RendererBase<T> {
         throw MustachioResolutionError(node.keySpan.message(
             "Failed to resolve '${e.name}' on ${e.contextType} while resolving "
             '$remainingNames as a property chain on any types in the context '
-            "chain: $contextChainString, after first resolving '${names.first}' "
-            'to a property on $T'));
+            "chain: $contextChainString, after first resolving '$firstName' to "
+            'a property on $T'));
       }
-    } else if (parent != null) {
+    }
+
+    if (this is SimpleRenderer && _invisibleGetters.contains(names.first)) {
+      var type = context.runtimeType;
+      throw MustachioResolutionError(node.keySpan.message(
+          '$firstName is a getter on $type, which is not visible to Mustache. '
+          'To render $firstName on $type, make it visible to mustache; to '
```

Comment:
might be more specific to help with maintenance, e.g. `add to visibleTypes`

Reply:
done